Mathlib/Data/List/Basic.lean

```diff
@@ -1348,6 +1359,11 @@ theorem take_one_drop_eq_of_lt_length' {l : List α} {n : ℕ} (h : n < l.length
 #align list.take_one_drop_eq_of_lt_length List.take_one_drop_eq_of_lt_length'
 
 #align list.ext List.ext
+theorem ext_get? : ∀ {l₁ l₂ : List α}, (∀ n, l₁.get? n = l₂.get? n) → l₁ = l₂ := List.ext
```

Comment:
Here, the style guide rule "hypotheses left of colon" applies: l₁, l₂ and `∀ n, l₁.get? n = l₂.get? n` should become hypotheses, so the conclusion is only `l₁ = l₂`.

Reply:
Also, please put a blank line before and after this theorem.

Reply:
Does "hypotheses left of colon" really apply? The condition "if the proof starts by introducing these variables" of the style guild is not satisfied (although I don't understand the dec_trivial example there)

Reply:
Good point - you don't explicitly introduce new variables in the beginning, that's right.
I'm quite sure some variables are introduced at least implicitly - I would put the first left of the colon for sure. I just tried it out, the second one also works...

And now that I tried it, it seems that is quite literally the same statement as `List.ext` in `std`.
Another reviewer may comment on this, but I wonder if this lemma should be removed, as it already exists.

Reply:
Indeed, I added an alias for List.ext to make the naming more consistent with my new ext-Lemma. I also don't know if this should be removed.

Reply:
I would remove this, and call the next lemma `ext_get?`. That seems consistent with [List.ext_get](https://leanprover-community.github.io/mathlib4_docs/Std/Data/List/Lemmas.html#List.ext_get) to me.